src/mock_etcd.rs

```diff
@@ -0,0 +1,565 @@
+            );
+            assert_eq!(
+                MockEtcd::map_get(Arc::clone(&mock_etcd.map), range2.clone())
+                    .await
+                    .len(),
+                4
+            );
+
+            // Test delete
+            let mut delete_no_exist = crate::etcd::DeleteRangeRequest::new();
+            delete_no_exist.set_key(vec![0_u8]);
+            delete_no_exist.set_range_end(vec![]);
+
+            let mut delete_one_key = crate::etcd::DeleteRangeRequest::new();
+            delete_one_key.set_key(vec![1_u8, 1_u8, 1_u8]);
+            delete_one_key.set_range_end(vec![]);
+            // delete range
+            let mut delete_range = crate::etcd::DeleteRangeRequest::new();
+            delete_range.set_key(vec![0_u8, 0_u8, 0_u8]);
+            delete_range.set_range_end(vec![0_u8, 1_u8, 0_u8]);
+            // delete all
+            let mut delete_all = crate::etcd::DeleteRangeRequest::new();
+            delete_all.set_key(vec![0_u8]);
+            delete_all.set_range_end(vec![0_u8]);
+
+            assert_eq!(
+                MockEtcd::map_delete(Arc::clone(&mock_etcd.map), delete_no_exist.clone())
+                    .await
+                    .len(),
+                0
+            );
+            assert_eq!(
+                {
+                    let kv =
+                        MockEtcd::map_delete(Arc::clone(&mock_etcd.map), delete_one_key.clone())
+                            .await;
+                    kv.get(0).unwrap().get_value().to_owned()
+                },
+                vec![1_u8, 1_u8, 1_u8]
+            );
+            assert_eq!(mock_etcd.map.read().await.len(), 7);
+            assert_eq!(
+                MockEtcd::map_delete(Arc::clone(&mock_etcd.map), delete_range.clone())
+                    .await
+                    .len(),
+                2
+            );
+            assert_eq!(mock_etcd.map.read().await.len(), 5);
+            assert_eq!(
+                MockEtcd::map_delete(Arc::clone(&mock_etcd.map), delete_all.clone())
+                    .await
+                    .len(),
+                5
+            );
+            assert_eq!(mock_etcd.map.read().await.len(), 0);
+        });
+    }
+    fn e2e_test() {
+        let etcd_service = crate::etcd_grpc::create_kv(MockEtcd::new());
+        let mut etcd_server = grpcio::ServerBuilder::new(Arc::new(Environment::new(1)))
+            .register_service(etcd_service)
+            .bind("127.0.0.1", 8888)
+            .build()
+            .unwrap_or_else(|_| panic!("failed to build etcd server"));
+        etcd_server.start();
+
+        smol::future::block_on(Compat::new(async {
+            let endpoints = vec!["http://127.0.0.1:8888".to_owned()];
+            let client = Client::connect(ClientConfig {
+                endpoints,
+                auth: None,
+                tls: None,
+            })
+            .await
+            .unwrap_or_else(|_| panic!("failed to connect to etcd server"));
```

Comment:
do not ignore error here, show the error content in panic info
`|err| panic!("failed to connect to etcd server, the error is: {}", err)`
same for all